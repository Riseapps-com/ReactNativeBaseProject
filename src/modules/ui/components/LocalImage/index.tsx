import React from 'react';
import { Image } from 'react-native';

import { useImages } from '~assets';

import type { Images } from '~assets';
import type { ImageProps, ImageStyle, StyleProp } from 'react-native';

export type LocalImageProps = Omit<ImageProps, 'source'> & {
  width?: number;
  height?: number;
  source: keyof Images;
};

const LocalImage: React.FC<LocalImageProps> = props => {
  const { width, height, source, style, ...restProps } = props;
  const imageStyles: StyleProp<ImageStyle> = [!!width && { width }, !!height && { height }, style];
  const images = useImages();
  const localSource = images[source];

  return <Image {...restProps} accessibilityLabel={source} source={localSource} style={imageStyles} />;
};

export default LocalImage;
