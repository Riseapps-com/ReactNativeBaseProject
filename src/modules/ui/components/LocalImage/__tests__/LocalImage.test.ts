import { renderComponent } from '~modules/tests';

import LocalImage from '../index';

import type { LocalImageProps } from '../index';

const renderLocalImage = (props: LocalImageProps) => renderComponent(LocalImage, props);

describe('ui', () => {
  describe('<LocalImage />', () => {
    it('renders <LocalImage />', () => {
      const localImage = renderLocalImage({
        source: 'flag',
      });

      expect(localImage.getByA11yLabel('flag')).toBeTruthy();
    });

    it('renders custom width and height', () => {
      const localImage = renderLocalImage({
        source: 'flag',
        width: 100,
        height: 200,
      });

      expect(localImage.getByA11yLabel('flag')).toHaveStyle({ width: 100, height: 200 });
    });
  });
});
