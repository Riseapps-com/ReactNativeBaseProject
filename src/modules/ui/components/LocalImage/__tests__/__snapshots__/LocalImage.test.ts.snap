// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <LocalImage /> renders <LocalImage /> 1`] = `
<Image
  accessibilityLabel="flag"
  source={
    Object {
      "testUri": "../../../assets/images/countries/flag.png",
    }
  }
  style={
    Array [
      false,
      false,
      undefined,
    ]
  }
/>
`;
